### PR TITLE
Update ResultToken API to optionally return token log probability 

### PR DESCRIPTION
**Pull request by wenxindongwork**
This PR updates the ResultToken class with an additional `log_prob` attribute which stores token log probabilities. This attribute is set to None by default. 

MaxEngine is updated to populate this attribute when `return_log_prob` is set to `True` in global config. (https://github.com/AI-Hypercomputer/maxtext/pull/1626)

### CODE REVIEW

**Review by mitalisi — COMMENTED** [review]
We dont always have to return logits. Would be better to have a flag for it.

**Review by jyj0w0 — APPROVED** [review]

**Review by jyj0w0 — APPROVED** [review]

**Comment by wenxindongwork** [comment]
This PR could not get merged. Perhaps because I'm using a fork. Trying this PR with a branch instead (https://github.com/AI-Hypercomputer/JetStream/pull/260)
